path_secrets.go

```diff
@@ -0,0 +1,94 @@
+package kubesecrets
+
+import (
+	"context"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const secretsPrefix = "secret_name"
+
+func pathSecrets(b *secretsReaderBackend) *framework.Path {
+	return &framework.Path{
+		Pattern: framework.MatchAllRegex(secretsPrefix),
+
+		Fields: map[string]*framework.FieldSchema{
+			"secret_name": {
+				Type:        framework.TypeString,
+				Description: "Specifies the name of the kubernetes secret.",
+				Query:       true,
+				Required:    true,
+			},
+			"namespace": {
+				Type:        framework.TypeString,
+				Description: "Specifies the name of the kubernetes secret namespace.",
+				Query:       true,
+				Required:    true,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation: b.handleRead,
+		},
+		HelpDescription: pathInvalidHelp,
+		ExistenceCheck:  b.handleExistenceCheck,
+	}
+}
+
+func (b *secretsReaderBackend) handleRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	secretName := data.Get("secret_name").(string)
+	namespace := data.Get("namespace").(string)
+	b.Logger().Info("In handleRead() secretName: " + secretName + ", namespace: " + namespace)
+
+	if secretName == "" {
+		resp := logical.ErrorResponse("Missing secret name")
+		return resp, nil
+	}
+
+	if namespace == "" {
+		resp := logical.ErrorResponse("Missing secret namespace")
+		return resp, nil
+	}
+
+	fetchedData, err := b.KubeSecretReader.GetSecret(secretName, namespace, b.Logger())
```

Comment:
Would it make sense to pass `ctx` here instead of creating a new one? Maybe clientgo respects cancels (or will respect some day). Just a thought.